File: src/test/java/fr/adrienbrault/idea/symfony2plugin/tests/doctrine/intention/DoctrineRepositoryClassConstantIntentionTest.java
package fr.adrienbrault.idea.symfony2plugin.tests.doctrine.intention;

import com.jetbrains.php.lang.PhpFileType;
import fr.adrienbrault.idea.symfony2plugin.tests.SymfonyLightCodeInsightFixtureTestCase;

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 *
 * @see fr.adrienbrault.idea.symfony2plugin.doctrine.intention.DoctrineRepositoryClassConstantIntention
 */
public class DoctrineRepositoryClassConstantIntentionTest extends SymfonyLightCodeInsightFixtureTestCase {

    public void setUp() throws Exception {
        super.setUp();
        myFixture.configureFromExistingVirtualFile(myFixture.copyFileToProject("classes.php"));
    }

    public String getTestDataPath() {
        return "src/test/java/fr/adrienbrault/idea/symfony2plugin/tests/doctrine/intention/fixtures";
    }

    public void testIntentionIsAvailable() {
        assertIntentionIsAvailable(PhpFileType.INSTANCE,
            "<?php\n /** @var $foo \\Doctrine\\Common\\Persistence\\ManagerRegistry */\n $foo->getRepository('foo<caret>')",
            "Doctrine: use class constant"
        );

        assertIntentionIsAvailable(PhpFileType.INSTANCE,
            "<?php\n /** @var $foo \\Doctrine\\Common\\Persistence\\ObjectManager */\n $foo->getRepository('foo<caret>')",
            "Doctrine: use class constant"
        );

        assertIntentionIsAvailable(PhpFileType.INSTANCE,
            "<?php\n /** @var $foo \\Doctrine\\Common\\Persistence\\ObjectManager */\n $foo->find('foo<caret>')",
            "Doctrine: use class constant"
        );
    }
}

File: src/test/kotlin/fr/adrienbrault/idea/symfony2plugin/tests/doctrine/intention/DoctrineRepositoryClassConstantIntentionTest.kt
package fr.adrienbrault.idea.symfony2plugin.tests.doctrine.intention

import com.jetbrains.php.lang.PhpFileType
import fr.adrienbrault.idea.symfony2plugin.tests.SymfonyLightCodeInsightFixtureTestCase

/**
 * @author Daniel Espendiller <daniel@espendiller.net>
 *
 * @see fr.adrienbrault.idea.symfony2plugin.doctrine.intention.DoctrineRepositoryClassConstantIntention
 */
class DoctrineRepositoryClassConstantIntentionTest : SymfonyLightCodeInsightFixtureTestCase() {
    override fun setUp() {
        super.setUp()
        myFixture.configureFromExistingVirtualFile(myFixture.copyFileToProject("classes.php"))
    }

    override fun getTestDataPath(): String {
        return "src/test/java/fr/adrienbrault/idea/symfony2plugin/tests/doctrine/intention/fixtures"
    }

    fun testIntentionIsAvailable() {
        assertIntentionIsAvailable(
            PhpFileType.INSTANCE,
            "<?php\n /** @var \$foo \\Doctrine\\Common\\Persistence\\ManagerRegistry */\n \$foo->getRepository('foo<caret>')",
            "Doctrine: use class constant",
        )

        assertIntentionIsAvailable(
            PhpFileType.INSTANCE,
            "<?php\n /** @var \$foo \\Doctrine\\Common\\Persistence\\ObjectManager */\n \$foo->getRepository('foo<caret>')",
            "Doctrine: use class constant",
        )

        assertIntentionIsAvailable(
            PhpFileType.INSTANCE,
            "<?php\n /** @var \$foo \\Doctrine\\Common\\Persistence\\ObjectManager */\n \$foo->find('foo<caret>')",
            "Doctrine: use class constant",
        )
    }
}
